app/controllers/signatures/index.js

```diff
@@ -200,27 +202,47 @@ export default class SignaturesIndexController extends Controller {
   }
 
   createSignFlow = task(async () => {
-    if (this.selectedPieces.length) {
-      await Promise.all(this.selectedPieces.map(async (piece) => {
-        const decisionActivity = await this.getDecisionActivity(piece);
+    try {
+      if (this.selectedPieces.length) {
+        await Promise.all(this.selectedPieces.map(async (piece) => {
+          const decisionActivity = await this.getDecisionActivity(piece);
+          await this.signatureService.createSignFlow(
+            piece,
+            decisionActivity,
+            this.signers,
+            this.approvers,
+            this.notificationAddresses
+          );
+        }));
+      } else if (this.piece) {
         await this.signatureService.createSignFlow(
-          piece,
-          decisionActivity,
+          this.piece,
+          this.decisionActivity,
           this.signers,
           this.approvers,
           this.notificationAddresses
         );
-      }));
-    } else if (this.piece) {
-      await this.signatureService.createSignFlow(
-        this.piece,
-        this.decisionActivity,
-        this.signers,
-        this.approvers,
-        this.notificationAddresses
+      }
+      this.closeSidebar();
+      await this.router.refresh(this.router.routeName);
+      if (this.selectedPieces.length > 1) {
+        this.toaster.success(
+          this.intl.t('documents-were-sent-to-signinghub'),
+          this.intl.t('successfully-started-sign-flows')
+        );
+      } else {
+        this.toaster.success(
+          this.intl.t('document-was-sent-to-signinghub'),
+          this.intl.t('successfully-started-sign-flow')
+        );
+      }
+    } catch {
+      this.closeSidebar();
+      await this.router.refresh(this.router.routeName);
```

Comment:
closing and refreshing executed on fail and success > `finally` block ?

Reply:
Perhaps, but the problem is that the toasts need to be created after we refresh, otherwise we show them briefly and they get lost during the refresh.

If you set it in a `finally` block, you also need to show the toast in the finally block, and you need to track the state for that (error or success, many or one).

Reply:
Perhaps, but the problem is that the toasts need to be created after we refresh, otherwise we show them briefly and they get lost during the refresh.

If you set it in a `finally` block, you also need to show the toast in the finally block, and you need to track the state for that (error or success, many or one).

Reply:
keep as-is